pdf/pdfcanvas.cpp

```diff
@@ -267,22 +278,48 @@ void PDFCanvas::layout()
     update();
 }
 
+static qreal dist2FromRect(const QRectF &rect, const QPoint &point)
+{
+    qreal dist = 0.;
+
+    if ((qreal)point.x() < rect.x()) {
+        dist += (rect.x() - (qreal)point.x()) * (rect.x() - (qreal)point.x());
+    } else if ((qreal)point.x() > rect.x() + rect.width()) {
+        dist += (rect.x() + rect.width() - (qreal)point.x())
+                * (rect.x() + rect.width() - (qreal)point.x());
+    }
+    if ((qreal)point.y() < rect.y()) {
+        dist += (rect.y() - (qreal)point.y()) * (rect.y() - (qreal)point.y());
+    } else if ((qreal)point.y() > rect.y() + rect.height()) {
+        dist += (rect.y() + rect.height() - (qreal)point.y())
+                * (rect.y() + rect.height() - (qreal)point.y());
+    }
+
```

Comment:
No, as far as I understand, one cannot use the Manhattan distance here. I mean, computing the Manhattan distance between the tap position and the link centre won't give good results. 

Indeed, the iso-lines for this distance are point-centred diamond-shapes (*i.e.* rotated square of 45°, for the usual distance, the iso-lines are point-centred spheres). Links can be (are often in fact) elongated boxes in x direction. Diamond-shapes are not matching these elongated shapes (very bad ascii art):
```
Manhattan
     /\        
    / . \   (should be square-shape)
    \  /
     \/
dist2FromBox
  /----------------------------\ (should be rounded corners in ASCII art also)
  |  link with very long text  |
  \----------------------------/
```
With the given function, the distance iso-lines correspond to something like a margin around each link box.